src/NLog/LayoutRenderers/PerformanceCounterLayoutRenderer.cs

```diff
@@ -71,13 +71,33 @@ public class PerformanceCounterLayoutRenderer : LayoutRenderer
         /// Gets or sets the name of the performance counter instance (e.g. this.Global_).
         /// </summary>
         /// <docgen category='Performance Counter Options' order='10' />
-        public string Instance { get; set; }
+        public string Instance
+        {
+            get { return _instance?.Text; }
+            set
+            {
+                _instance = new SimpleLayout(value);
+                _fixedPerformanceCounter = null;
+                _performanceCounter = null;
+            }
+        }
+        private SimpleLayout _instance;
 
         /// <summary>
         /// Gets or sets the name of the machine to read the performance counter from.
         /// </summary>
         /// <docgen category='Performance Counter Options' order='10' />
-        public string MachineName { get; set; }
+        public string MachineName
+        {
+            get { return _machineName?.Text; }
+            set
+            {
+                _machineName = new SimpleLayout(value);
+                _fixedPerformanceCounter = null;
+                _performanceCounter = null;
```

Comment:
Please create a helper `ClearFields` (3 times needed), to set them all to `null`

Reply:
Fixed